src/blueprint/operators/const_field.rs

```diff
@@ -25,27 +28,120 @@ fn validate_data_with_schema(
+                    if m.is_const() {
+                        out.push(MustacheOrValue::Value(v.clone()));
+                    } else {
+                        out.push(MustacheOrValue::Mustache(m));
+                    }
+                }
+                Ok(ValueOrDynamic::MustacheArray(out))
+            }
+            Value::String(s) => {
+                let m = Mustache::parse(s.as_str())?;
+                if m.is_const() {
+                    Ok(ValueOrDynamic::Value(value.clone()))
+                } else {
+                    Ok(ValueOrDynamic::Mustache(m))
+                }
+            }
+            _ => Ok(ValueOrDynamic::Value(value.clone())),
+        }
+    }
+}
+
 pub fn compile_const(inputs: CompileConst) -> Valid<Expression, String> {
     let config_module = inputs.config_module;
     let field = inputs.field;
     let value = inputs.value;
     let validate = inputs.validate;
 
-    let data = value.to_owned();
-    match ConstValue::from_json(data.to_owned()) {
-        Ok(gql) => {
-            let validation = if validate {
-                validate_data_with_schema(config_module, field, gql)
+    let data = value;
+    match data {
+        ValueOrDynamic::Value(v) => match ConstValue::from_json(v.to_owned().to_owned()) {
+            Ok(gql) => {
+                let validation = if validate {
+                    validate_data_with_schema(config_module, field, gql)
+                } else {
+                    Valid::succeed(())
+                };
+                validation.map(|_| Literal(data.to_owned()))
+            }
+            Err(e) => Valid::fail(format!("invalid JSON: {}", e)),
+        },
+        ValueOrDynamic::MustacheObject(map) => {
+            let a = map.into_iter().filter(|(_, v)| !v.is_const()).count();
+            if a > 0 {
+                Valid::succeed(Literal(data.to_owned()))
             } else {
-                Valid::succeed(())
-            };
-            validation.map(|_| Literal(data))
+                let mut out = IndexMap::new();
+                for (k, v) in map {
+                    match v {
+                        MustacheOrValue::Mustache(_) => {
+                            unimplemented!("mustache in const object")
+                        }
+                        MustacheOrValue::Value(v) => {
+                            out.insert(k.clone(), ConstValue::from_json(v.to_owned()).unwrap());
+                        }
+                    }
+                }
+                let obj = ConstValue::Object(out);
+                let validation = if validate {
+                    validate_data_with_schema(config_module, field, obj)
+                } else {
+                    Valid::succeed(())
+                };
+                validation.map(|_| Literal(data.to_owned()))
+            }
         }
-        Err(e) => Valid::fail(format!("invalid JSON: {}", e)),
+        _ => Valid::succeed(Literal(data.to_owned())),
```

Comment:
Lot of tests are missing for the conversion.

Reply:
Validation if mustache is used in const is not implemented adding a todo